src/Compilers/CSharp/Portable/Symbols/DelegateCaches/DelegateCacheManager.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Threading;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.CSharp.Emit;
+using Microsoft.CodeAnalysis.Symbols;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    using ModuleScopedContainerCollection = ConcurrentDictionary<TypeSymbol, ModuleScopedDelegateCacheContainer>;
+
+    internal sealed class DelegateCacheManager : IComparer<ModuleScopedDelegateCacheContainer>
```

Comment:
The name of the class is confusing, it manages only module level containers, not all of them. Consider changing the name to reflect this.